src/main/java/com/thirdcc/webapp/service/impl/UserCCInfoServiceImpl.java

```diff
@@ -91,6 +99,83 @@ public UserCCInfoDTO save(UserCCInfoDTO userCCInfoDTO) {
             .map(this::clubFamilyDetails);
     }
 
+    @Override
+    @Transactional(readOnly = true)
+    public List<UserCCInfoDTO> getUserCCInfoByUserId(Long userId) {
+        log.debug("Request to get UserCCInfo of User : {}", userId);
+        List<UserCCInfoDTO> userCCInfoDTOList = userCCInfoRepository
+            .findAllByUserId(userId, Pageable.unpaged())
+            .map(userCCInfoMapper::toDto)
+            .getContent();
+        return getFullUserCCInfoList(userCCInfoDTOList, userId);
+    }
+
+    /**
+     * User CC Fish Level:
+     * First Year Session is "Junior Fish"
+     * Second Year Session is "Senior Fish"
+     * Later Year Session until Present is "Elder Fish"
+     *
+     * @param userCCInfoDTOList
+     * @param userId
+     * @return List of user CC Info Sorted by Year Session
+     */
+    private List<UserCCInfoDTO> getFullUserCCInfoList(List<UserCCInfoDTO> userCCInfoDTOList, Long userId) {
+        List<String> yearSessionWithCCFamilyRole = userCCInfoDTOList.stream()
+            .map(UserCCInfoDTO::getYearSession)
+            .collect(Collectors.toList());
+        List<UserCCInfoDTO> fullUserCCInfoDTOList = new ArrayList<>();
+        userUniInfoService.getUserUniInfoByUserId(userId)
+            .ifPresent(userUniInfoDTO -> {
```

Comment:
what if `Optional optional = userUniInfoService.getUserUniInfoByUserId(userId)` is `Optional.empty()`?
what will happen?

lengthy lambad function is hard to read. I prefer using 
```java
if (!optional.isPresent) {
// not found 
 return smtg;
}
Object obj = optional.get();
// do further processing if found
return obj;
```

Reply:
actually if Optional.empty() 
no need to do anything

Anyway I will change since there is a request

Reply:
resolved in d5253a35de87edf18588bc0c2bae3356b925f969